File: algorithm/src/main/java/baekjoon/p07785/Main.java
package baekjoon.p07785;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashSet;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();

        String[] inputs = new String[n];
        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.nextLine();
        }

        for (String solve : solve(inputs)) {
            System.out.println(solve);
        }
    }

    private static List<String> solve(String[] inputs) {
        HashSet<String> hashSet = new HashSet<>();

        for (String input : inputs) {
            String[] arr = input.split(" ");
            if (arr[1].equals("enter")) {
                hashSet.add(arr[0]);
            } else {
                hashSet.remove(arr[0]);
            }
        }

        List<String> ret = new ArrayList<>(hashSet);
        Collections.sort(ret);
        Collections.reverse(ret);

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p07785/Main.kt
package baekjoon.p07785

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    val inputs = List(n) { readLine()!! }
    println(solve(inputs).joinToString("\n"))
}

private fun solve(inputs: List<String>): List<String> {
    val hashSet = HashSet<String>()
    for (input in inputs) {
        val arr = input.split(" ")
        when (arr[1]) {
            "enter" -> hashSet.add(arr[0])
            "leave" -> hashSet.remove(arr[0])
        }
    }

    return hashSet.sortedDescending()
}